### DISCUSSION TITLE
AWS NLB config via ingress controller

### DISCUSSION THREAD

**Discussion by sbellary** [nginx/kubernetes-ingress#5096] [Q&A]
Hi
Does nginx ingress controller allow setting up AWS NLB with a pre-defined security group? If so, how can i do it?
I'm using 1.9 version of ingress controller and have it configured to create an internet facing AWS NLB. I would like to control LB access via a security group. 
Is there a way for me to configure this with helm chart itself? Trying to avoid additional manual/scripting to do this

Appreciate your responses/recommendations on this. 

Thanks

**Comment by j1m-ryan** [comment]
Hi @sbellary, it sounds like you might be using the [community nginx ingress project](https://github.com/kubernetes/ingress-nginx), who are currently on version `v1.9.x`. We did have a `v1.9.0` and `v1.9.1`, but these are 2 and half years old. Before diving into this I would like to confirm that it is this project (`nginxinc/kubernetes-ingress`)  that you are using and not `kubernetes/ingress-nginx`

**Reply by sbellary** [reply]
Hi @j1m-ryan .. you are right, i'm on community project (kubernetes/ingress-nginx).

**Reply by sbellary** [reply]
If I get the ability to configure SGs on LBs that ingress controller creates, switching to `nginxinc/kubernetes-ingress` should not be an issue for me

**Reply by j1m-ryan** [reply]
Looking [here](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.6/guide/service/annotations/#security-groups), theres probably two annotations you care about
`service.beta.kubernetes.io/aws-load-balancer-security-groups: sg-xxxx, nameOfSg1, nameOfSg2`  
`service.beta.kubernetes.io/aws-load-balancer-manage-backend-security-group-rules: "false"`

You can add these to the service annotations in helm

**Reply by sbellary** [reply]
I had tried those but it doesn't seem to make any difference. I didn't see that taking effect, neither specified SG is getting attached to NLB nor the additional rules getting attached to the node SG.
That's why i was wondering if its related to the ingress controller chart i'm using
